src/mongrove/core.clj

```diff
@@ -186,6 +190,7 @@
                   (.writeConcern (get write-concern-map write-concern))
                   (.readPreference (get read-preference-map read-preference))
                   (.retryWrites retry-writes)
+                  (.credential credential)
```

Comment:
The credential needs to be conditionally handled. Initialization without a username and password is failing.